packages/blockchain-extension/extension/commands/addEnvironmentCommand.ts

```diff
@@ -135,18 +135,21 @@ export async function addEnvironment(): Promise<void> {
         let addedAllNodes: boolean;
 
         if (createMethod === UserInputUtil.ADD_ENVIRONMENT_FROM_OPS_TOOLS) {
-            addedAllNodes = await vscode.commands.executeCommand(ExtensionCommands.EDIT_NODE_FILTERS, fabricEnvironmentEntry, true, UserInputUtil.ADD_ENVIRONMENT_FROM_OPS_TOOLS);
+            addedAllNodes = await vscode.commands.executeCommand(ExtensionCommands.EDIT_NODE_FILTERS, fabricEnvironmentEntry, true, createMethod);
         } else {
-            addedAllNodes = await vscode.commands.executeCommand(ExtensionCommands.IMPORT_NODES_TO_ENVIRONMENT, fabricEnvironmentEntry, true, UserInputUtil.ADD_ENVIRONMENT_FROM_NODES);
+            addedAllNodes = await vscode.commands.executeCommand(ExtensionCommands.IMPORT_NODES_TO_ENVIRONMENT, fabricEnvironmentEntry, true, createMethod);
         }
-
         if (addedAllNodes === undefined) {
             await fabricEnvironmentRegistry.delete(fabricEnvironmentEntry.name);
             return;
-        }
-
-        if (addedAllNodes) {
-            outputAdapter.log(LogType.SUCCESS, 'Successfully added a new environment');
+        } else if (addedAllNodes) {
+            const environment: FabricEnvironment = new FabricEnvironment(fabricEnvironmentEntry.name);
```

Comment:
If the nodes are added from an environment then you shouldn't let it create with no nodes, only if its from ops tools

Reply:
If no nodes are added in a non-OpsTools environment in the importNodesToEnvironment command, this will result in an error being thrown, ie, it is only possible to be in this code path if it is an opsTool environment.